source/extensions/filters/network/rbac/config.cc

```diff
@@ -76,10 +78,15 @@ Network::FilterFactoryCb
 RoleBasedAccessControlNetworkFilterConfigFactory::createFilterFactoryFromProtoTyped(
     const envoy::extensions::filters::network::rbac::v3::RBAC& proto_config,
     Server::Configuration::FactoryContext& context) {
-  validateRbacRules(proto_config.rules());
-  validateRbacRules(proto_config.shadow_rules());
+  if (proto_config.has_rules()) {
```

Comment:
I have a question, why are not doing the validation of the matcher in the function `createFilterFactoryFromProtoTyped` like the current code?

Reply:
In the `MatcherEngineImpl`, we will validate and populate the match tree. I just do not want to take the match tree and action context from `FilterConfigFactory` to `FilterConfig` and to `createEngine` and finally to `MatcherEngineImpl`.

Reply:
No worry, It will throw in the config path.